packages/browser-wallet/src/popup/shared/utils/account-helpers.ts

```diff
@@ -49,11 +49,11 @@ export function usePrivateKey(accountAddress: string | undefined): string | unde
 
     const privateKey = useMemo(() => {
         // We don't throw errors on missing credentials or identities, as they might just be loading.
-        if (!accountAddress || !credential || !identity) {
+        if (!accountAddress || !credential || !identity || seedPhrase.state !== 'hasData') {
             return undefined;
         }
 
-        return ConcordiumHdWallet.fromHex(seedPhrase, getNet(network))
+        return ConcordiumHdWallet.fromHex(seedPhrase.data, getNet(network))
             .getAccountSigningKey(identity.providerIndex, identity.index, credential.credNumber)
             .toString('hex');
     }, [credential?.credId, seedPhrase, identity?.index]);
```

Comment:
```suggestion
    }, [credential?.credId, seedPhrase.state, identity?.index]);
```
oder was?

Reply:
The value is present in the `data` subfield when `state === 'hasData'`.

Reply:
Sure.
But don't we have problems with having objects in the dependency array?
And the assumption is that the data won't change?
Maybe this is fine actually

Reply:
I misread where the comment was placed. I will change it to state, and yes the assumption is that data won't change here.